### PR TITLE
feature: add custom theme

### PR DESCRIPTION
**Pull request by vdtdev**
AOS2-16 [Implement Custom Theme](https://bitovi.atlassian.net/browse/AOS2-16?atlOrigin=eyJpIjoiMmRjNmYyNTJkMDFlNDhhMmJhMWZiYWVkMGM1YWM5ZjkiLCJwIjoiaiJ9)
- [x] Add custom theme support
- [x] Add dark mode support

Dark Mode:
image

Light Mode:
image

### CODE REVIEW

**Review by krivanek06 — APPROVED** [review]

File: src/theme.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// ==[Dark Theme]==
+$dark-primary: mat.define-palette(mat.$blue-palette, 200);
+$dark-accent: mat.define-palette(mat.$red-palette, 200, 100, 400);
+
+$dark-theme: mat.define-dark-theme(
+	(
+		color: (
+			primary: $dark-primary,
+			accent: $dark-accent,
```
Comment (m-thompson-code) [inline src/theme.scss:30]: :0 no warning?

**Comment by m-thompson-code** [comment]
tests are failing:

Screen Shot 2022-11-16 at 10 07 45 AM

Don't worry about all the other errors. I'll create a PR to clean everything out. Moving forward, PR's should be blocked if they throw errors during tests. I'll add a checklist to the PR template to make sure we run our tests before trying to open our PR's

**Comment by m-thompson-code** [comment]
Looks like using the partial transparent border-color overlaps with the background of the div, which creates a noticeable border. I think a workaround is to set the border-color to transparent. This works since the border overlaps the background of the div:

Screen Shot 2022-11-16 at 10 09 56 AM

Screen Shot 2022-11-16 at 10 10 08 AM

**Comment by m-thompson-code** [comment]
On light mode, the border for container seems to have become lighter:

Screen Shot 2022-11-16 at 10 16 51 AM

Screen Shot 2022-11-16 at 10 17 00 AM

I think the original border-color for white mode is better: rgba(0,0,0,.2)

**Review by m-thompson-code — CHANGES_REQUESTED** [review]
Everything looks pretty good code-wise. There's a few issues I ran into when testing:

1. tests are failing
2. light-mode container border-color should be rgba(0,0,0,.2)
3. dark-mode container border-color should be transparent

File: src/theme.scss (hunk 1 above)
Comment (vdtdev) [reply src/theme.scss:30]: oops

Comment (vdtdev) [reply src/theme.scss:31]: Fixed!

**Review by m-thompson-code — APPROVED** [review]
awesome~ now I have dark mode finally :3

**Pull request merged**